tools/container/wrapper/wrapper.go

```diff
@@ -0,0 +1,115 @@
+/**
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"errors"
+	"io/fs"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+func main() {
+	program, err := os.Executable()
+	if err != nil {
+		log.Fatalf("failed to get executable: %v", err)
+	}
+	if isRuntimeWrapper(program) && !isNvidiaModuleLoaded() {
+		log.Println("nvidia driver modules are not yet loaded, invoking runc directly")
+		program, err := exec.LookPath("runc")
+		if err != nil {
+			log.Fatalf("failed to find runc: %v", err)
+		}
+		argv := []string{"runc"}
+		argv = append(argv, os.Args[1:]...)
+		execve(program, argv, os.Environ())
+	}
+	argv := makeArgv(program)
+	envv := makeEnvv(program)
+	execve(program+".real", argv, envv)
```

Comment:
Question: Is there any reason to not call `syscal.Exec()` here directly?

Reply:
I just put it in a function to have a single point of error handling (not that fatal logging is much handling). I can inline this.